### PR TITLE
Captain Megaphone

### PR DESCRIPTION
**Pull request by Flaborized**
[FEAT]

## About the PR 

Adds a megaphone to the captains locker. Holding the megaphone in your hand and speaking will make your maptext larger and your voice louder! (Larger in chatbox). Since you're the most important person on the station you can now have the easiest to read text to match! (Or have it be stolen and be shamed as it's used against you) 
Has an emag effect 👁️ (Non-gameplay-altering)

Compatible with the red and blue captain medals (if you hold it in your hand while activating)

## Why's this needed? 

I think it's a fun cute lil feature! Wanted to make this for a while and have finally gotten around to it. Helps the captain feel like they have a "commanding presence". Really just for fun roleplaying purposes though. Can be var edited to make the text much larger if an admin pleases.

Only one on the station so hopefully not a very annoying item either! I tried to balance it being noticeable but not obnoxious.

## Changelog 

```changelog
(u)Flaborized
(+)The captain's locker has been stocked with a megaphone to help them get their finer points across.
```

### CODE REVIEW

**Comment by DisturbHerb** [comment]
At one point many years ago, one of the HoSii did suggest adding such a megaphone to the Armory. Not sure if that'd fit, but the option is there. 👀

**Comment by muzcl** [comment]
Having the emag effect cause ear damage would be nice, not many things deal ear damage in this game I can think of. If its too overpowered like that, you could make it break after a few uses to balance it.

File: code/procs/mobprocs/chatprocs.dm [hunk 1]
```diff
@@ -466,6 +466,13 @@
 	if (src.get_brain_damage() >= 60)
 		speechverb = pick("says","stutters","mumbles","slurs")
 
+	if (src.find_type_in_hand(/obj/item/megaphone))
+		var/obj/item/megaphone/megaphone = src.find_type_in_hand(/obj/item/megaphone)
+		if	(megaphone.makes_you_quieter)
```
Comment (Azrun) [inline code/procs/mobprocs/chatprocs.dm:471]:
```suggestion
		if (megaphone.makes_you_quieter)
```

File: code/mob/living.dm [hunk 2]
```diff
@@ -1104,10 +1104,16 @@
 		else
 			maptext_color = src.last_chat_color
 
+		var/popup_style = src.speechpopupstyle
+
+		if (src.find_type_in_hand(/obj/item/megaphone))
+			var/obj/item/megaphone/megaphone = src.find_type_in_hand(/obj/item/megaphone)
+			popup_style = "font-size: [megaphone.maptext_size]px;"
+
```
Comment (Azrun) [inline code/mob/living.dm:1112]: Should this be `+=` otherwise it will replace the bioeffects?
/datum/bioEffect/speech/comic

**Comment by Flaborized** [comment]
Undrafting this since I added the things I think I can add to this. Made the text bold and green (with a slight dark green outline) to better help it stand out and make it look less... thin. Added blue and red variants that you can get by holding it and activating the correct medals. 

As for the sound, I tried for a while to mix something good in audacity but I can't really make something passable! I'm ok adding it without a sound and I can't *personally* provide one I'm happy with in the time frame I wanna work on this for :sheltersweat:

new look:

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**